### ISSUE TITLE
[Server Process Manager] MRI code path should not be mandatory

### ISSUE THREAD

**Issue by nicolasbrossard** [aces/Loris#4750]
Config setting MRI code path should not be mandatory to use the server process manager. Right  now if you try to access the module and you do not have that Config setting, you get an HTTP 500 error.

**Comment by johnsaigle** [comment]
I think mriCodePath is where LORIS-MRI is installed. Shouldn't it be mandatory to use server process manager, since it's used only for image uploads?

**Comment by nicolasbrossard** [comment]
The genomic browser also uses the server process manager.

**Comment by johnsaigle** [comment]
Does it? I had no idea. 

The README for server processes manager says:
> ## Interactions with LORIS
The imaging uploader module is currently the only LORIS module capable of
launching an asynchronous process, namely the execution of the MRI processing
pipeline on a successfully uploaded scan archive. Consequently, the result table
will only show these type of processes.

Is this out of date?

**Comment by nicolasbrossard** [comment]
I believe so. The thing is, I cannot find any reference to the server process manager in Xavier's genomic browser module...I am positive he uses it for his genomic stuff. We can wait a bit for him to come back so he can shed some light on this.

**Comment by ridz1208** [comment]
@xlecours can you provide an answer ?

**Comment by xlecours** [comment]
@nicolasbrossard The GenomicBrowser do not use the ServerProcessManager. But there is developpement for the Phantom Pipeline that does use the SPM. 

As for the mriCodePath requirment, it should be enforce when someone wants to launch the mri_upload task; not when he wants to look at launched tasks. But since its merged and since the only way a task could exists is when the path is set, I don't think this is blocking.

**Comment by cmadjar** [comment]
@driusan Should we remove the 21.0 release tag then? It could be a change implemented in later releases if I understand correctly.

**Comment by xlecours** [comment]
I think that https://github.com/aces/Loris/blob/main/modules/server_processes_manager/php/module.class.inc#L56 validateConfig can be skipped at this point. Creating on instantiating a MRIUpload server process will simply fail with a Configuration Exception is MRICodePath is not set.

**Comment by nicolasbrossard** [comment]
I agree with @xlecours 's last comment. As of 2024Sep17, this should be the fix that would allow this ticket to be closed.
